### PR TITLE
`Displacement`

### PR DESCRIPTION
**Pull request by Dillon-Roller**

### CODE REVIEW

File: src/board.rs [hunk 1]
```diff
@@ -71,29 +72,31 @@ impl Board {
...
+        Self::is_in_bounds(m.to)?;
+        self.is_piece_some(m.from)?;
+        if self.moves.contains(&m) {
             Ok(())
         } else {
             Err(ChessError::InvalidMove)
         }
     }
 
     pub fn move_piece(&mut self, from: Position, to: Position) -> ChessResult<()> {
```
Comment (DanielAlbl) [inline src/board.rs:86]: Can we take a `Move` here?

File: src/board.rs [hunk 2]
```diff
@@ -177,7 +189,7 @@ impl Board {
         for y in 0..8 {
             for x in 0..8 {
                 if let Some(piece) = self.squares[y][x] {
-                    if piece.get_player() == self.player {
+                    if self.player == piece.get_player() {
```
Comment (DanielAlbl) [inline src/board.rs:192]: Lol

File: src/board.rs (hunk 1 above)
Comment (Dillon-Roller) [reply src/board.rs:86]: Which exactly?

**Review by DanielAlbl — APPROVED** [review]

File: src/board.rs (hunk 1 above)
Comment (DanielAlbl) [reply src/board.rs:86]: Its fine

**Pull request merged**